src/codegen/compile_enhanced.ml

```diff
@@ -2311,10 +2311,13 @@ module Closure = struct
     Tagged.load_field env (Int64.add header_size i)
 
   let store_data env i =
-    let (set_closure_data, get_closure_data) = new_local env "closure_data" in
-    set_closure_data ^^
-    Tagged.load_forwarding_pointer env ^^
-    get_closure_data ^^
+    (if G.to_instr_list (Tagged.load_forwarding_pointer env) <> [] then
+       let set_closure_data, get_closure_data = new_local env "closure_data" in
+       set_closure_data ^^
+       Tagged.load_forwarding_pointer env ^^
+       get_closure_data
+     else
+       G.nop) ^^
```

Comment:
```suggestion
     let (set_closure_data, get_closure_data) = new_local env "closure_data" in
    set_closure_data ^^
    Tagged.load_forwarding_pointer env ^^
    get_closure_data ^^
```
Revert - compile_enhance is incremental_gc only so the forwarding is always required.

Reply:
Oh, yeah, missed that!

Reply:
Done in 8821ac888e6e87445e9fcbf5ea6c65dd4bbeaaa8